Source/Renderer/FramebufferManager.js

```diff
@@ -0,0 +1,126 @@
+import Framebuffer from "./Framebuffer.js";
+import PixelDatatype from "./PixelDatatype.js";
+import Renderbuffer from "./Renderbuffer.js";
+import RenderbufferFormat from "./RenderbufferFormat.js";
+import Sampler from "./Sampler.js";
+import Texture from "./Texture.js";
+import Color from "../Core/Color.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import PixelFormat from "../Core/PixelFormat.js";
+
+/**
+ * Creates a wrapper object around a framebuffer and its resources.
+ *
+ * @private
+ * @constructor
+ */
+function FramebufferManager(options) {
+  options = defaultValue(options, defaultValue.EMPTY_OBJECT);
+  this._depthStencil = defaultValue(options.depthStencil, false);
+  this._useHdr = false;
+
+  this._framebuffer = undefined;
+  this._colorRenderbuffer = undefined;
+  this._colorTexture = undefined;
+  this._depthStencilRenderbuffer = undefined;
+  this._depthStencilTexture = undefined;
+}
+
+FramebufferManager.prototype.isDirty = function (width, height, hdr) {
+  hdr = defaultValue(hdr, false);
+  var colorTexture = this._colorTexture;
+  return (
+    !defined(colorTexture) ||
+    colorTexture.width !== width ||
+    colorTexture.height !== height ||
+    this._useHdr !== hdr
+  );
+};
+
+FramebufferManager.prototype.update = function (
```

Comment:
For now, I'd say the current approach is manageable. `depthTexture` probably doesn't need to be passed in because it can be gotten from `context.depthTexture` and I don't think any of the calling code needs to override that.

Reply:
A couple places use the `context.depthTexture` check like `SceneFramebuffer` and `GlobeTranslucenyFramebuffer`, but others don't.. I'll revisit this when I add the FramebufferManager to more classes.

Reply:
There are also some classes that can make use of this option to force a depth-stencil renderbuffer instead of texture (i.e. `PickFramebuffer._fb`).